subprojects/messaging/src/main/java/org/gradle/internal/serialize/DefaultSerializerRegistry.java

```diff
@@ -28,8 +28,11 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class DefaultSerializerRegistry implements SerializerRegistry {
+    private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
     private final Map<Class<?>, Serializer<?>> serializerMap = new TreeMap<Class<?>, Serializer<?>>(new Comparator<Class<?>>() {
```

Comment:
Is there any policy about adding `@GuardedBy` annotations around the fields that require synchronization?  I've always found that helpful - it makes it harder to miss the need to maintain a synchronization policy when adding new methods that touch the fields.  This is a pretty short class though, not counting the inner classes, maybe that would be overkill.

Reply:
I could do this, yea